heron/spi/src/java/com/twitter/heron/spi/packing/InstanceId.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.twitter.heron.spi.packing;
+
+/**
+ * Encapsulates the unique components of an Instance identifier.
+ */
+public final class InstanceId {
```

Comment:
No, it's actually no longer needed. It was only used to be parsed by the packing classes and the executor. Now that it's contents are part of InstanceId, it's no longer needed. There is still a convenient task identifier generated and used by the executor so simplify things like log and pid files and to avoid naming collisions, but that's it. From [the executor](https://github.com/twitter/heron/blob/master/heron/executor/src/python/heron_executor.py#L382):
```
instance_id = "container_%s_%s_%s" % (str(self.shard), component_name, str(global_task_id))
```